### PR TITLE
play feature

### PR DESCRIPTION
**Pull request by umair313**

### CODE REVIEW

File: songs/templates/app/playlist.html [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        </div>
+        <div>
+          <ion-icon name="heart" id="likebtn" dataset="5" class="like_icon liked"></ion-icon>
+        </div>
+    </div>
+    {% endfor %}
+ {% else %}
+ {% endif %}
+</div>
+<script type="module" src="https://unpkg.com/ionicons@4.5.10-0/dist/ionicons/ionicons.esm.js"></script>
```
Comment (jawadchughtai6) [inline songs/templates/app/playlist.html]: Add this in base

File: songs/urls.py [hunk 2]
```diff
@@ -1,10 +1,18 @@
 from django.urls import path
-from .views import ListView, ArtistListView, AlbumListView, TrackDetailView, AlbumDetailView, ArtistDetailView
+from .views import (ListView, ArtistListView, AlbumListView, LikeSongsPlaylistView,
+                    TrackDetailView, AlbumDetailView, ArtistDetailView,
+                    LikeTrack, UnlikeTrack)
 urlpatterns = [
-    path('list/', ListView.as_view(), name='listing'),
-    path('listing/artists/', ArtistListView.as_view(), name='artist_list'),
-    path('listing/albums/', AlbumListView.as_view(), name='album_list'),
+    path('', ListView.as_view(), name='listing'),
```
Comment (jawadchughtai6) [inline songs/urls.py]: Have this as "list/" and remove the "search/" route

File: songs/views.py [hunk 3]
```diff
@@ -36,13 +36,52 @@ class TrackDetailView(generic_views.DetailView):
...
+    def get(self, request, *args, **kwargs):
+        user = request.user
+        tracks = user.playlists.filter(name="Liked Songs").first().tracks.all()
+        return render(request, 'app/playlist.html', context={"tracks": tracks})
+
+
+def LikeTrack(request, track_id):
+    user = request.user
+    playlist = user.playlists.filter(name='Liked Songs').first()
+    track = Track.objects.filter(id=track_id).first()
```
Comment (jawadchughtai6) [inline songs/views.py]: Read up on get() and get_object_or_404() methods

File: songs/views.py [hunk 4]
```diff
@@ -36,13 +36,52 @@ class TrackDetailView(generic_views.DetailView):
...
+    object = playlist.tracks.filter(id=track_id).first()
+    if not object:
```
Comment (jawadchughtai6) [inline songs/views.py]: use ```if object:``` instead

File: songs/views.py [hunk 5]
```diff
@@ -36,13 +36,52 @@ class TrackDetailView(generic_views.DetailView):
...
+        PlaylistTrack.objects.create(playlist=playlist, track=track)
+        return JsonResponse({'message': True})
+    else:
+        return JsonResponse({'message': False})
+
+
+def UnlikeTrack(request, track_id):
+    user = request.user
+    playlist = user.playlists.filter(name='Liked Songs').first()
+    track = Track.objects.filter(id=track_id).first()
```
Comment (jawadchughtai6) [inline songs/views.py]: Use get_object_or_404 here

**Review by jawadchughtai6 — CHANGES_REQUESTED** [review]

File: songs/views.py (hunk 5 above)
Comment (umair313) [reply songs/views.py]: now using  get_object_or_404

File: songs/views.py (hunk 4 above)
Comment (umair313) [reply songs/views.py]: remvoed `if object`

**Pull request merged**